include/lsst/afw/geom/SpanSetFunctorGetters.h

```diff
@@ -0,0 +1,261 @@
+
+    // No need to increment, make a no-op function
+    void increment() {}
+
+    T get() const {return _const;}
+ private:
+    T _const;
+};
+
+template <typename T, int N, int C>
+class ImageNdGetter {
+    // Getter class to manage iterating though an ndarray which is interpreted as a 2D image
+ public:
+    using Reference = typename ndarray::Array<T, N, C>::Reference::Reference;
+
+    ImageNdGetter(ndarray::Array<T, N, C> const & array, geom::Point2I xy0): _array(array), _xy0(xy0) {}
+
+    void checkExtents(Box2I const & bbox, int area) const {
+        // If the bounding box lays outside the are of the image, throw an error
+        geom::Box2I arrayBBox(_xy0, _xy0+geom::Extent2I(_array.template getSize<1>(),
+                                                        _array.template getSize<0>()));
+        if (!arrayBBox.contains(bbox)) {
+            throw LSST_EXCEPT(lsst::pex::exceptions::OutOfRangeError,
+                              "SpanSet bounding box lands outside array");
+        }
+    }
+
+    void setSpan(Span const & span) {
+        auto _iterY = _array.begin() + (span.getY() - _xy0.getY());
+        _iterX = (*_iterY).begin() + (span.getMinX() - _xy0.getX());
+    }
+
+    void increment() { ++_iterX; }
+
+    Reference get() {
+        return *_iterX;}
+
+ private:
+    ndarray::Array<T, N, C>  _array;
+    geom::Point2I _xy0;
+    typename ndarray::Array<T, N, C>::Reference::Iterator _iterX;
+};
+
+template <typename T>
+class FlatNdGetter {
+    // Getter class to manage iterating though an ndarray which is interpreted as a 2D image
+ public:
+    using Reference = typename ndarray::Array<T, 1, 1>::Reference;
+
+    explicit FlatNdGetter(ndarray::Array<T, 1, 1> const & array): _array(array) {
+        _iter = _array.begin();
+    }
+
+    void checkExtents(Box2I const & bbox, int area) {
+        // If the area of the array is greater than the size of the array, throw an error
+        // as the iteration dimensions will not match
+        auto shape = _array.getShape();
+        if (area > static_cast<int>(shape[0])) {
+            throw LSST_EXCEPT(lsst::pex::exceptions::OutOfRangeError,
+                              "SpanSet has dimensionality greater than array");
+        }
+    }
+
+    void setSpan(Span const & span) {}
+
+    void increment() { ++_iter; }
+
+    Reference get() const {
+        return *_iter;}
+
+ private:
+    ndarray::Array<T, 1, 1>  _array;
+    typename ndarray::Array<T, 1, 1>::Iterator _iter;
+};
+
+template <typename T>
+FlatNdGetter<T> makeGetter(FlatNdGetter<T> & getter) {
+    // This function simply passes through any FlatNdGetter passed to it
+    return getter;
+}
+
+template <typename T, int inA, int inB>
+ImageNdGetter<T, inA, inB> makeGetter(ImageNdGetter<T, inA, inB> & getter) {
+    // This function simply passes though any imageNdGetter passed to it
+    return getter;
+}
+
+template <typename T>
+ImageNdGetter<T, 2, 1> makeGetter(lsst::afw::image::Image<T> & image) {
```

Comment:
It is redundant because your non-const overload doesn't do anything different -- it doesn't modify the image, and it doesn't return a getter with extra permissions. In fact, the call to the `ImageNdGetter` constructor const-qualifies the object returned by `image.getArray()` regardless of whether `image` is const or not, so you would have exactly the same effect if you just had the const method.